File: src/_as/02/genome_key16.ts
// SSoT: src/ontology/memory/genome_key16.md
import { LOGIC_OFFSET } from "../01/mod";

@inline
export function genome_key16(idx: i32): i32 {
const ptr = LOGIC_OFFSET + (idx << 3);
const b0 = load<u8>(ptr) as i32;
const b1 = load<u8>(ptr + 1) as i32;
return (b0 << 8) | b1;
}

File: src/00/sigma_core/src/ontology_gen/02/genome_key16.rs
// SSoT: file:///Users/s0fractal/OMEGA/src/ontology/memory/genome_key16.md
#![allow(unused_imports)]
use super::super::L01::*;

pub fn genome_key16(state: &SigmaState, idx: i32) -> i32 {
    let b0 = state.matrix.logic[idx as usize][0] as i32;
    let b1 = state.matrix.logic[idx as usize][1] as i32;
    (b0 << 8) | b1
}
